scripts/entity/components/attack_components/base_attack_component.gd

```diff
@@ -1,11 +1,55 @@
 class_name BaseAttackComponent extends Node
 
+## Occurs when the target has changed
+signal on_target_change(previous_target, current_target)
+
+## The Area2D that the character locks onto the closest target.
+@export var lock_on_area: Area2D
+
+## The node to compare against the targets.
+@export var measurement_position: Node2D
+
+## The time between target updates in seconds.
+@export var time_between_target_updates_in_seconds: float
+
+## Used to continue a coroutine for updating the target.
+var _update_targets: bool = true
+
+## The current target that is closest in the area.
+var _current_target: Node2D = null
 
 # Called when the node enters the scene tree for the first time.
 func _ready() -> void:
-	pass # Replace with function body.
+	on_target_change.connect(target_changed)
+	update_closest_target()
+
+
+## Checks each target in the lock on area and then determines which is the closest. Emits an on_target_change signal if the target has changed.
+func update_closest_target():
+	while _update_targets:
```

Comment:
This should work a lot better, but I think having the functionality of "if there is no targets, then add target based on Area2D enter trigger" would be good, which would also allow us to not need to run it indefinitely - only if there are possible active targets.

Reply:
I updated it. Its not nearly as clean now but it might function a little better